include/appbase/application_base.hpp

```diff
@@ -131,10 +131,10 @@ class application_base {
          bool more = true;
 
          while (more || io_serv.run_one()) {
-            if (is_quiting())
-               break;
             try {
                io_serv.poll(); // queue up any ready; allowing high priority item to get into the queue
+               if (io_serv.stopped())
```

Comment:
I wonder if we should also change the `is_quitting()` function to return `io_serv.stopped()`, and remove the `is_quitting` bool?

Reply:
It uses a mutex instead of an `atomic`.